compiler/name-gen.cpp

```diff
@@ -35,6 +35,17 @@ bool is_array_suitable_for_hashing(VertexPtr vertex) {
   return vertex->type() == op_array && CheckConst::is_const(vertex);
 }
 
+bool is_object_suitable_for_hashing(VertexPtr vertex) {
+  bool ss = vertex->type() == op_define_val && vertex.as<op_define_val>()->value()->type() == op_func_call && vertex->const_type == cnst_const_val;
+  return ss;
+}
+
+std::string gen_const_object_name(const VertexAdaptor<op_define_val> &def) {
```

Comment:
We don't pass `VertexAdaptor`, like `VertexPtr`/`FunctionPtr`/etc. by a const reference, it's a POD 8-byte type
(I see, that it's done in `gen_const_array_name()`, it's strange; it you just have done by analogy, may ignore this comment)